supplier_app/views.py

```diff
@@ -220,6 +223,10 @@ def form_valid(self, forms):
             address = forms['address_form'].save(commit=False)
             address.taxpayer = taxpayer
             address.save()
+            contact = forms['contact_form'].save(commit=False)
+            contact.taxpayer = taxpayer
+            contact.address = address
+            contact.save()
```

Comment:
This `try` block is getting bigger, first we can split it in different methods like `create_taxpayer`, `set_taxpayer_entities`, `set_taxpayer_address`, etc and then we can think in an abstraction that has the responsibility to create a taxpayer and left the django view simpler (just receive a request, validate the form and redirect)

Reply:
Thanks, for the review. It's updated

Reply:
Check commit 'Refactoring form_valid'

Reply:
Nice! just solve the @ricardocaif-eb comments!